torch/_subclasses/fake_impls.py

```diff
@@ -429,6 +431,15 @@ def nonzero(fake_mode, func, arg):
 
             if not has_free_symbols(arg.numel()):
                 maxval = int(arg.numel())
+            else:
+                prod_node = math.prod(arg.shape).node
+                prod_range = bound_sympy(
+                    prod_node.expr, prod_node.shape_env.var_to_range
+                )
```

Comment:
I guess I won't force you to do this, but we should think about factoring this into some sort of helper so we don't have to be grabbing `.node` directly here

Reply:
Which dir would I put a helper like this?